app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -108,10 +106,10 @@ public void onStart() {
         // Default Database rules do not allow unauthenticated reads, so we need to
         // sign in before attaching the RecyclerView adapter otherwise the Adapter will
         // not be able to read any data from the Database.
-        if (!isSignedIn()) {
-            signInAnonymously();
-        } else {
+        if (isSignedIn()) {
             attachRecyclerViewAdapter();
+        } else {
```

Comment:
@puf I write my PRs with the philosophy that small improvements as we go along are better than constantly submitting side PRs for this kind of change. However, I see where you are coming from and removed these changes from this PR.